app/screens/receive-bitcoin-screen/receive-btc.tsx

```diff
@@ -453,53 +253,69 @@ const ReceiveBtc = () => {
         }
       }
     }
-  }, [paymentFullUri])
 
-  const togglePaymentLayer = () => {
-    setInvoice(null)
-    setPaymentLayer((currentPaymentLayer) => {
-      return currentPaymentLayer === TYPE_LIGHTNING_BTC
-        ? TYPE_BITCOIN_ONCHAIN
-        : TYPE_LIGHTNING_BTC
-    })
-  }
-
-  const lastHash = useLnUpdateHashPaid()
-  const [invoicePaid, setInvoicePaid] = useState(false)
-  useEffect(() => {
-    if (lastHash === invoice?.paymentHash) {
-      setInvoicePaid(true)
+    return {
+      copyToClipboard,
+      share,
     }
-  }, [invoice?.paymentHash, lastHash])
+  }, [invoice, LL])
 
-  const satAmountInUsd = convertCurrencyAmount({
-    amount: satAmount,
-    from: "BTC",
-    to: "USD",
-  })
+  if (!invoiceDetails || !setAmount) {
+    return <></>
+  }
 
-  if (showAmountInput) {
-    const usdAmountInSats = Math.round(
-      convertCurrencyAmount({
-        amount: usdAmount ?? 0,
-        from: "USD",
-        to: "BTC",
-      }),
-    )
+  const {
+    unitOfAccountAmount,
+    settlementAmount,
+    convertPaymentAmount,
+    memo,
+    invoiceType,
+  } = invoiceDetails
+
+  const toggleAmountCurrency =
+    unitOfAccountAmount &&
+    (() => {
+      const newAmountCurrency =
+        unitOfAccountAmount.currency === WalletCurrency.Usd
+          ? WalletCurrency.Btc
+          : WalletCurrency.Usd
+      setAmount(convertPaymentAmount(unitOfAccountAmount, newAmountCurrency))
+    })
+  const toggleInvoiceType = () => {
+    const newInvoiceType =
+      invoiceDetails.invoiceType === InvoiceType.Lightning
+        ? InvoiceType.OnChain
+        : InvoiceType.Lightning
+    setInvoiceType(newInvoiceType, true)
+  }
+  const btcAmount = settlementAmount
+  const usdAmount =
+    unitOfAccountAmount && convertPaymentAmount(unitOfAccountAmount, WalletCurrency.Usd)
+  const setAmountsWithBtc = (sats: number) => {
+    setAmount({
+      amount: sats,
+      currency: WalletCurrency.Btc,
+    })
+  }
+  const setAmountsWithUsd = (dollars: number | null) => {
+    setAmount({
+      amount: Math.round(Number(dollars) * 100),
+      currency: WalletCurrency.Usd,
+    })
+  }
 
-    const validAmount =
-      (amountCurrency === "BTC" && satAmount !== null) ||
-      (amountCurrency === "USD" && usdAmount !== null)
+  if (showAmountInput && unitOfAccountAmount && btcAmount && usdAmount) {
+    const validAmount = Boolean(invoiceDetails.unitOfAccountAmount.amount)
 
     return (
       <View style={[styles.inputForm, styles.container]}>
         <View style={styles.currencyInputContainer}>
           <View style={styles.currencyInput}>
-            {amountCurrency === "BTC" && (
+            {unitOfAccountAmount.currency === WalletCurrency.Btc && (
               <>
                 <FakeCurrencyInput
```

Comment:
why do we have this name for the input?

Reply:
Its a poorly named component from a library that we use that we want to get rid of. `Fake` in this context means that it is not a real React TextInput component but a simulated or fake TextInput component.